projects/table-builder/src/lib/classes/table-store.ts

```diff
@@ -180,7 +193,48 @@ export class TableStore extends ComponentStore<TableState> {
     if(sorted.length === 0) {
       sorted = this.createPreSort(metaData);
     }
-    return {...state, metaData, sorted};
+    const order = this.initializeOrder(state, metaData);
+    return {...state, metaData, sorted, userDefined:{...state.userDefined,order:order}};
   });
 
+  private initializeOrder = (state:TableState,mds: Dictionary<MetaData>) : Dictionary<number> => {
+    const viewableMetaDataArr = Object.values(mds).sort((a,b)=>a.order-b.order).filter(a => a.fieldType !== FieldType.Hidden);
+    const userDefinedOrder = state.userDefined.order;
+    let userDefinedOrderArr = Object.entries(userDefinedOrder);
+
+    if( thereIsSavedOrder() && viewableMetaAddedSinceLastSave()){
```

Comment:
in my opinion.
In a case where the developer added a column just remove the user defined array.
Like I said, there's no good way to know what the user meant.
He could mean column 2 he could mean before column 3 or after column 1.